main/lsp/notifications/sorbet_workspace_edit.h

```diff
@@ -10,10 +10,8 @@ class SorbetWorkspaceEditTask final : public LSPDangerousTypecheckerTask {
     absl::Notification startedNotification;
     std::unique_ptr<Timer> latencyCancelSlowPath;
     std::unique_ptr<SorbetWorkspaceEditParams> params;
-    // Caches the file hashes computed for `canPreempt` to avoid recomputing them later.
-    mutable std::vector<core::FileHash> cachedFileHashesOrEmpty;
-    // Caches the fast path decision made with cachedFileHashesOrEmpty.
-    // Is only valid if `cachedFileHashesOrEmpty` is not empty.
+    // Caches the fast path decision for the provided update. Becomes invalidated when the update changes.
```

Comment:
I used to use `cachedFileHashesOrEmpty.empty()` to decide if the cached fast path decision was valid.

Now I use an explicit boolean.

In a follow-up PR, a trinary value will replace `cachedFastPathDecision`, and it will be explicitly `NOT_DECIDED` when it hasn't been determined.

Reply:
Seems reasonable.  FWIW, it also would seem reasonable to me to make `cachedFastPathDecision` be an `optional<bool>` (assuming that compiles correctly/efficiently) if you don't want to invent a trinary type.